plugins/email_service/app/controllers/email_service/configsets_controller.rb

```diff
@@ -1,13 +1,14 @@
 module EmailService
   class ConfigsetsController < ::EmailService::ApplicationController
     before_action :restrict_access
-
+    before_action :check_user_creds_roles
```

Comment:
using the policy engine you wouldn't need this before checks

Reply:
adjusted in further commits.